Sources/MongoSwift/MongoConnectionString.swift

```diff
@@ -8,6 +9,22 @@ public struct MongoConnectionString: Codable, LosslessStringConvertible {
     /// - SeeAlso: https://datatracker.ietf.org/doc/html/rfc3986#section-2.2
     fileprivate static let forbiddenUserInfoCharacters = [":", "/", "?", "#", "[", "]", "@"]
 
+    private enum Names {
```

Comment:
maybe we should consider giving the enum a raw type of `String`, and making these static properties enum cases instead? 

it might make things a little safer in the big switch on option names below, since you could initialize an enum instance from the option name and switch on that. would also I think make the TODO at the bottom to error on unknown options easy

Reply:
Weird, I originally made this a struct but it looks like the formatter changed it into an enum 🤔 Switching to a proper enum worked out nicely though, updated

Reply:
ah, yeah generally an empty enum is considered the "right" way to do namespacing: https://github.com/nicklockwood/SwiftFormat/blob/master/Rules.md#enumnamespaces 

we don't do this much in the driver but we did use it to namespace our error types so they are all prefixed with `MongoError`